recipes-web: Update iotivity-node and iot-rest-api-server to latest

nagineni opened the pull request:
Pull in the iotivity-node version 1.2.0-2. 

Signed-off-by: Sudarsana Nagineni <sudarsana.nagineni@intel.com>

nagineni commented:
@mikko @poussa

poussa commented:
+1

mythi commented on recipes-web/iot-rest-api-server/iot-rest-api-server.bb:
```diff
@@ -27,8 +27,7 @@ SYSTEMD_AUTO_ENABLE_${PN} = "disable"
 USERADD_PARAM_${PN} = "\
---system --home ${localstatedir}/lib/empty \
---no-create-home --shell /bin/false \
+--system --create-home --shell /bin/false \
```
the commit message says iotivity-node installs to home but did you mean to say iot-rest-api-server?

with useradd-staticids.bblass used in Ostro, the home dir is picked from ostro-passwd and is defined as ```/var/lib/empty```. Was the problem that iot-rest-api-server was not able to create that dir but now that it gets created here it's OK? Perhaps explain the problem in the commit message.

nagineni replied:
@mythi Updated the commit message.

Issue was that, iotivity-node instantiated by rest api server was failing to create configuration files under ${HOME}/.iotivity-node directory when the server started under non-root user (i.e restful). The reason for that was the restful user was created without a home directory. So, I changed --no-create-home option to --create-home to create user's home directory.

Commit:
iot-rest-api-server.bb: Create user's home directory

iotivity-node instantiated by rest api server had permission
issues in creating a configuration file with persistent state
related information under ${HOME}/.iotivity-node directory
when the server started under non-root user (i.e restful).

The reason for the failure was that the restful user was created
without home directory. So, this patch fixes the problem by
creating user's home directory with --create-home option.

Signed-off-by: Sudarsana Nagineni <sudarsana.nagineni@intel.com>

Commit:
recipes-web: Update iotivity-node and iot-rest-api-server to latest version

Pull in the latest iotivity-node and iot-rest-api-server.

Signed-off-by: Sudarsana Nagineni <sudarsana.nagineni@intel.com>

mythi commented:
On Fri, Feb 3, 2017 at 11:15 AM, Sudarsana Nagineni  wrote:


> @mythi <https://github.com/mythi> Updated the commit message.
>
> Issue was that, iotivity-node instantiated by rest api server was failing
> to create configuration files under ${HOME}/.iotivity-node directory when
> the server started under non-root user (i.e restful). The reason for that
> was the restful user was created without a home directory. So, I changed
> --no-create-home option to --create-home to create user's home directory.
>
>
You should be getting /var/lib/empty with this change.

nagineni commented:
> You should be getting /var/lib/empty with this change.

Yes, that is what we need for now to solve the issue. I will change the dir to /home/restful in ostro-passwd in a separate PR later.

mythi commented:
+1
